File: 190. Reverse Bits.py
# 190. Reverse Bits

class Solution:
    def reverseBits(self, n: int) -> int:
        ans = 0
        for _ in range(32):
            ans <<= 1
            ans += n & 1
            n >>= 1

        return ans
    
    
# class Solution:
#     def reverseBits(self, n: int) -> int:
#         res = 0
#         for i in range(32):
#             digit = n >> i & 1
#             res = digit << (31 - i) | res

#         return res

File: 190. Reverse Bits.cpp
// 190. Reverse Bits


class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        long res {0};
        long digit;

        for (int i = 31; i >=0; i--) {
            digit = n & 1;
            res += digit << i;
            n >>= 1;
        }
        return res;
    }
};